syntax = "proto3";

// Not necessary for Python but should still be declared to avoid name collisions
// in the Protocol Buffers namespace and non-Python languages
package protoblog;

// Style guide prefers prefixing enum values instead of surrounding
// with an enclosing message

enum Actions {
    ADD_ITEM = 0;
    SET_VALUE = 1;
    DELETE_KEY = 2;
    DELETE_DB = 3;
}

message DBAction {
    Actions action_type = 1;
    optional string key = 2;
    optional string value = 3;
}

message DBInfo {
    string db_name = 1;
    int32 client_id = 2;
}

message GetDB {
    bytes db_file = 1;
}

message GetDBHash {
    string db_hash = 1;
}

enum DBState {
    NOT_SYNCED = 0;
    SYNCED = 1;
}

message GetDBState {
    DBState state = 1;
}

